### ISSUE TITLE
Machine Learning CLI

### ISSUE THREAD

**Issue by molbazsa** [evoRegression/Regression#102]
# Possible features of the CLI:

## Training data
- [ ] `generate` command

## Training
- [x] `train` command
- [x] Specify network architecture (e.g. `--layers`, `--activationfunc`)
- [x] Load numerical data (from json, e.g. `train data --input`, support gzip as well)
  - [x] Create DataSet
  - [x] Import / Export DataSet
  - [x] Compress DataSet
  - [x] `train data` command
- [ ] Load image data (from pictures using the ImageProcessor component, e.g. `train images --input`)
- [x] Specify parameters (e.g. `--batchsize`, `--learningrate`, `--epochs`)
- [x] Specify output file (e.g. `--output`)
- [ ] Plot error (e.g. `--error`)

## Propagation for testing
- [ ] `propagate` command

## Usage
- `.\LinearRegressionBackend.exe train data "$env:TESTTMP\dataset.json" "$env:TESTTMP\trained_network.json.gz" --layers 2 2 2 --activationfunc Sigmoid`
- `.\LinearRegressionBackend.exe train data "$env:TESTTMP\dataset.json.gz" "$env:TESTTMP\trained_network.json.gz" --layers 2 2 2 --activationfunc Sigmoid --batchsize 10`
